### ISSUE TITLE
Notification system

### ISSUE THREAD

**Issue by imphil** [librecores/librecores-web#143]
**Goal:** inform users about events happening on the page, e.g. invitations to new organizations, outstanding requests of any kind, etc.

Don't send out only emails, but also display notifications on the page itself. Let users enable/disable notifications, and choose how they would like to be informed about it (email, on the page, not at all).

First step is a design document for this feature.

**Comment by eliaskousk** [comment]
I agree, a design document is a must for this feature. Implementing this until the next meeting is probably impossible (excuses!) but I will try to at least have the design document and #142 ready for a code review.

**Comment by aquibbaig** [comment]
We can work out a methodology but first I want to know what events do we want to notify the users for?

**Comment by aquibbaig** [comment]
Here are a few things that can be proposed.
Steps involved in implementing a **notification system** for Librecores(with prior reference to my proposal):
- [ ] Discuss designs for the notifications entity(e.g. fields like date, eventName, etc.)
- [ ] Design of changes in database structure for storing user notifications
- [ ] Generating notifications from events and storing it in the database for a user
- [ ] Fetching notifications from the database to the UI
- [ ] Displaying notifications

We need to discuss which bundles to use or implement this from scratch, whatever will be a viable option needs to be decided according to website requirements, also adhering to my proposal

**Comment by aquibbaig** [comment]
@imphil is this the right way?

**Comment by imphil** [comment]
Please
- Start a new issue "Design of the notification system", add it to your Project view on Github (https://github.com/librecores/librecores-web/projects/5)
- Create a Google Doc linked from this issue.

Describe in this document:
- Where are notifications coming from? (notification sources) 
- What data is associated with them?
- What output mechanisms exist? (notification sinks) Examples are email and in-site notifications.
- What kind of routing and configuration is required to connect "sources" and "sinks"?

Note: All of this design decisions comes from a *user* perspective. It's not about how you implement it, it's about what the user sees or can see when interacting with the system.

Once we agree on a design in your Google Doc, copy it over to the issue and we can start with the next step: how should it be implemented?
